# Statistik-Labor Testat Aufgabe 3
***Hinweise:***
- Bitte überprüfen Sie Ihre Resultate vor der Abgabe Ihre Ergebnisse mit den Teilergebnissen aus der Datei ***Teilergebnisse_xy.txt***.
- Die Unterlagen sind im **pdf-Format** in Moodle hochgeladen abzugeben. Richtige Lösungen werden nicht mehr an Sie zurückgegeben. Eine Abgabe per Email ist nicht möglich.
- Bitte füllen Sie das jeweilige **Deckblatt** mit aus und geben es mit Ihrer Lösung zusammen ab.

***Aufgabenstellung:***<br>
In dieser Aufgabe wird die Lotterie „KENO“ untersucht, die die staatlichen Lottogesellschaften täglich (von Montag bis Samstag) anbieten. Bei dieser Lotterie besteht ein Tippfeld aus 70 Zahlen. Ein:e Teilnehmer:in an der Lotterie kann selbst entscheiden, wie viele dieser Zahlen er/sie ankreuzt (mindestens 2, höchstens 10) und welchen Betrag er/sie einsetzt (1, 2, 5 oder 10 Euro). Von den 70 Zahlen werden 20 Gewinnzahlen gezogen. Je nachdem, wie viele der Gewinnzahlen er/sie angekreuzt hatte, bekommt der/die Lotterieteilnehmer:in einen festen Geldbetrag ausgezahlt. Der Gewinnplan (Stand 01.01.2005) ist im Folgenden aufgelistet. (Es gibt Sonderregelungen für die jeweils höchsten Gewinnklassen bei 10 oder 9 getippten Zahlen; diese sollen nicht berücksichtigt werden und sind daher nicht hier aufgeführt.)

<table><tr>
<img src="https://drive.google.com/uc?id=19AUBO7UMNIHeCLUrULL-9gBY_WuODh5n" alt="Drawing" style="width: 400px;"/>
</tr><tr>
<td> <img src="https://drive.google.com/uc?id=1x2IijEgRFxBhZ4u3tF_jjLrpHqCMhKq_" alt="Drawing" style="width: 400px;"/> </td>
<td> <img src="https://drive.google.com/uc?id=1OkchOBHmd8B4yCESXvlz4orbVJj36Fag" alt="Drawing" style="width: 400px;"/> </td>
</tr><tr>
<td> <img src="https://drive.google.com/uc?id=1rOwpI4uG842uuew_fnQcxs1OisT3Wpje" alt="Drawing" style="width: 400px;"/> </td>
<td> <img src="https://drive.google.com/uc?id=1f8BLVyVaUh3SYmIATn8SB3uereJUeB9F" alt="Drawing" style="width: 400px;"/> </td>
</tr><tr>
<td> <img src="https://drive.google.com/uc?id=17SzdR7YXZxZ1RNn4v_KdwyV1XxD4_TtX" alt="Drawing" style="width: 400px;"/> </td>
<td> <img src="https://drive.google.com/uc?id=1suOwSx1Wudiow2Z314dSjhODOQypszgK" alt="Drawing" style="width: 400px;"/> </td>
</tr><tr>
<td> <img src="https://drive.google.com/uc?id=13-WILkeidHuGA9YCBnVyYOt5uJceWMWD" alt="Drawing" style="width: 400px;"/> </td>
<td> <img src="https://drive.google.com/uc?id=1MrYtnAJnSWLX81AMF6MNd3qR6Qpjdm0J" alt="Drawing" style="width: 400px;"/> </td>
</tr></table>

Ablesebeispiel: Sie haben auf dem Spielschein in einem Tippfeld 7 Zahlen angekreuzt; Ihr Einsatz beträgt $2\,€$. Wenn Sie unter Ihren 7 Zahlen genau 5 Richtige sind, erhalten Sie $2\cdot12\,€ = 24 €$ ausbezahlt (d. h. Ihr Gewinn beträgt $22\,€$). Wenn Sie stattdessen nur 3 Richtige haben, erhalten Sie nichts (d. h. Sie haben einen Verlust in Höhe Ihres Einsatzes von $2\,€$).

### a) Tippfeld mit 10 Kästchen 
Die Zufallsvariablen $X_{10}$ beschreibt den Gewinn eines:r Lotterieteilnehmer:in, der/die in einem Tippfeld 10 Kästchen ankreuzt und $2\,€$ einsetzt. Berechnen Sie Erwartungswert und Standardabweichung von $X_{10}$.

In [3]:
from scipy.stats import hypergeom

# Gewinne pro 1 Euro Einsatz
wins = [100000*2-2, 1000*2-2, 100*2-2, 15*2-2, 5*2-2, 2, -2, -2, -2, -2, 2]  

# Erstellung der Hypergeometrischen Verteilung
N, m, n = 70, 10, 20 #Parameter für die Hypergeometrische Verteilung,
Tippfeld = hypergeom(N, m, n)  #N=70 Gesamtanzahl der Zahlen, m=20 mögliche Zahlen, es werden 20 gezogen, n=10 Anzahl der Zahlen im Tippfeld

# Wahrscheinlichkeitsverteilung bei einem Tippfeld von 10
Tippfeld10 = [Tippfeld.pmf(i) for i in range(11)] #Array aus den einzelnen Wahrscheinlichkeiten, dass 10 richtig, 9..., 0 richtig

# Erwartungswert
erwartungswert = 0
for k in range(11): #11 da es ja 11 Möglichkeiten gibt also 0 richtige - 10 richtige sind 11 unterschiedliche Möglichkeiten
    erwartungswert += (wins[k] *Tippfeld10[10-k])
    
# Varianz
varianz = 0
for k in range(11):
    varianz += (wins[k]**2*Tippfeld10[10-k])
varianz -= erwartungswert**2


# Standardabweichung
standardabweichung = varianz**0.5


print('Erwartungswert:', erwartungswert.round(4))
print('Varianz:', varianz.round(4))
print('Standardabweichung:', standardabweichung.round(4))





Erwartungswert: -1.012
Varianz: 18735.7766
Standardabweichung: 136.8787


### b) Weitere Tippfelder
Berechnen Sie ebenso die Erwartungswerte und Standardabweichungen von $X_9, ..., X_2$, d. h. dem Gewinn bei Ankreuzen von 9 (bzw. 8, ..., 2) Kästchen in einem Tippfeld jeweils bei Einsatz von $2\,€$.

***Beispiellösung:***<br>
- Abgabe für a) und b) mit vollständiger Berechnung (nicht nur Endergebnisse)
- am liebsten in tabellarischer Form für $X_{10},...,X_2$

In [6]:
#Gewinne der Tippfelder 9-2
wins9 = [ 50000*2-2, 1000*2-2,  20*2-2,  5*2-2,     2,-2, -2, -2, -2,  2 ] 
wins8 = [ 10000*2-2,  100*2-2,  15*2-2,      2,     0,-2, -2, -2,  0 ] 
wins7 = [  1000*2-2,  100*2-2,  12*2-2,      0,    -2,-2, -2, -2 ]  
wins6 = [   500*2-2,   15*2-2,       2,      0,    -2,-2, -2 ]  
wins5 = [   100*2-2,    7*2-2,       2,     -2,    -2,-2 ] 
wins4 = [    22*2-2,        2,       0,     -2,    -2 ] 
wins3 = [    16*2-2,        0,      -2,     -2 ] 
wins2 = [     6*2-2,       -2,      -2 ]



#-------------------Tippfeld 9----------------------

# Erstellung der Hypergeometrischen Verteilung
N, m, n = 70, 9, 20 
Tippfeld = hypergeom(N, m, n)

# Wahrscheinlichkeitsverteilung Tippfeld 9
Tippfeld9 = [Tippfeld.pmf(i) for i in range(10)] 

# Erwartungswert 9
erwartungswert9 = 0
for k in range(10): 
    erwartungswert9 += (wins9[k] *Tippfeld9[9-k])
    
# Varianz
varianz9 = 0
for k in range(10):
    varianz9 += (wins9[k]**2*Tippfeld9[9-k])
varianz9 -= erwartungswert9**2

# Standardabweichung
standardabweichung9 = varianz9**0.5



#-------------------Tippfeld 8----------------------

# Erstellung der Hypergeometrischen Verteilung
N, m, n = 70, 8, 20 
Tippfeld = hypergeom(N, m, n)

# Wahrscheinlichkeitsverteilung Tippfeld 8
Tippfeld8 = [Tippfeld.pmf(i) for i in range(9)] 

# Erwartungswert 8
erwartungswert8 = 0
for k in range(9): 
    erwartungswert8 += (wins8[k] *Tippfeld8[8-k])
    
# Varianz
varianz8 = 0
for k in range(9):
    varianz8 += (wins8[k]**2*Tippfeld8[8-k])
varianz8 -= erwartungswert8**2

# Standardabweichung
standardabweichung8 = varianz8**0.5



#-------------------Tippfeld 7----------------------

# Erstellung der Hypergeometrischen Verteilung
N, m, n = 70, 7, 20 
Tippfeld = hypergeom(N, m, n)

# Wahrscheinlichkeitsverteilung Tippfeld 7
Tippfeld7 = [Tippfeld.pmf(i) for i in range(8)] 

# Erwartungswert 7
erwartungswert7 = 0
for k in range(8): 
    erwartungswert7 += (wins7[k] *Tippfeld7[7-k])
    
# Varianz
varianz7 = 0
for k in range(8):
    varianz7 += (wins7[k]**2*Tippfeld7[7-k])
varianz7 -= erwartungswert7**2

# Standardabweichung
standardabweichung7 = varianz7**0.5



#-------------------Tippfeld 6----------------------

# Erstellung der Hypergeometrischen Verteilung
N, m, n = 70, 6, 20 
Tippfeld = hypergeom(N, m, n)

# Wahrscheinlichkeitsverteilung Tippfeld 6
Tippfeld6 = [Tippfeld.pmf(i) for i in range(7)]

# Erwartungswert 6
erwartungswert6 = 0
for k in range(7): 
    erwartungswert6 += (wins6[k] *Tippfeld6[6-k])
    
# Varianz
varianz6 = 0
for k in range(7):
    varianz6 += (wins6[k]**2*Tippfeld6[6-k])
varianz6 -= erwartungswert6**2

# Standardabweichung
standardabweichung6 = varianz6**0.5



#-------------------Tippfeld 5----------------------

# Erstellung der Hypergeometrischen Verteilung
N, m, n = 70, 5, 20 
Tippfeld = hypergeom(N, m, n)

# Wahrscheinlichkeitsverteilung Tippfeld 5
Tippfeld5 = [Tippfeld.pmf(i) for i in range(6)]

# Erwartungswert 5
erwartungswert5 = 0
for k in range(6): 
    erwartungswert5 += (wins5[k] *Tippfeld5[5-k])
    
# Varianz
varianz5 = 0
for k in range(6):
    varianz5 += (wins5[k]**2*Tippfeld5[5-k])
varianz5 -= erwartungswert5**2

# Standardabweichung
standardabweichung5 = varianz5**0.5



#-------------------Tippfeld 4----------------------

# Erstellung der Hypergeometrischen Verteilung
N, m, n = 70, 4, 20 
Tippfeld = hypergeom(N, m, n)

# Wahrscheinlichkeitsverteilung Tippfeld 4
Tippfeld4 = [Tippfeld.pmf(i) for i in range(5)]

# Erwartungswert 4
erwartungswert4 = 0
for k in range(5): 
    erwartungswert4 += (wins4[k] *Tippfeld4[4-k])
    
# Varianz
varianz4 = 0
for k in range(5):
    varianz4 += (wins4[k]**2*Tippfeld4[4-k])
varianz4 -= erwartungswert4**2

# Standardabweichung
standardabweichung4 = varianz4**0.5



#-------------------Tippfeld 3----------------------

# Erstellung der Hypergeometrischen Verteilung
N, m, n = 70, 3, 20 
Tippfeld = hypergeom(N, m, n)

# Wahrscheinlichkeitsverteilung Tippfeld 3
Tippfeld3 = [Tippfeld.pmf(i) for i in range(4)]

# Erwartungswert 3
erwartungswert3 = 0
for k in range(4): 
    erwartungswert3 += (wins3[k] *Tippfeld3[3-k])
    
# Varianz
varianz3 = 0
for k in range(4):
    varianz3 += (wins3[k]**2*Tippfeld3[3-k])
varianz3 -= erwartungswert3**2

# Standardabweichung
standardabweichung3 = varianz3**0.5



#-------------------Tippfeld 2----------------------

# Erstellung der Hypergeometrischen Verteilung
N, m, n = 70, 2, 20 
Tippfeld = hypergeom(N, m, n)

# Wahrscheinlichkeitsverteilung Tippfeld 2
Tippfeld2 = [Tippfeld.pmf(i) for i in range(3)]

# Erwartungswert 2
erwartungswert2 = 0
for k in range(3): 
    erwartungswert2 += (wins2[k] *Tippfeld2[2-k])
    
# Varianz
varianz2 = 0
for k in range(3):
    varianz2 += (wins2[k]**2*Tippfeld2[2-k])
varianz2 -= erwartungswert2**2

# Standardabweichung
standardabweichung2 = varianz2**0.5



#--------------Tabellarische Darstellung---------------

import pandas as pd

#Tabellarische Darstellung
data = {
    'Tippfeld': ['x10', 'x9', 'x8', 'x7', 'x6', 'x5', 'x4', 'x3', 'x2'],
    'Erwartungswert': [erwartungswert, erwartungswert9, erwartungswert8, erwartungswert7, erwartungswert6, erwartungswert5, erwartungswert4, erwartungswert3, erwartungswert2],
    'Varianz': [varianz, varianz9, varianz8, varianz7, varianz6, varianz5, varianz4, varianz3, varianz2],
    'Standardabweichung':[standardabweichung, standardabweichung9, standardabweichung8, standardabweichung7, standardabweichung6, standardabweichung5, standardabweichung4, standardabweichung3, standardabweichung2]
}

data = pd.DataFrame(data)

data.round(4)




Tippfeld  Erwartungswert     Varianz  Standardabweichung
0      x10         -1.0120  18735.7766            136.8787
1       x9         -0.9991  26218.0757            161.9200
2       x8         -1.0212   5358.7205             73.2033
3       x7         -1.0087    331.7899             18.2151
4       x6         -1.0051    301.3534             17.3595
5       x5         -1.0020     56.0134              7.4842
6       x4         -1.0111     11.2623              3.3559
7       x3         -0.9865     20.9925              4.5818
8       x2         -1.0559     10.4379              3.2308

### c) Gewinnchance-Maximierung
Wie viele Kästchen pro Tippfeld sollte ein:e KENO-Spieler:in ankreuzen, der/die den Erwartungswert seines/ihres Gewinns maximieren möchte?

#### Antwort:
- Er muss 3 Kästchen pro Tippfeld ankreuzen um seine Gewinnchance zu maximieren, da dort der Erwartungswert am höchsten ist

### d) Individual Aufgabe
Beantworten Sie die in Datei **sr_aufg_3d_xy.txt** im Unterverzeichnis **Endziffer_xy** gestellte Frage (*xy = Endziffern Ihrer Matrikelnummer*). Welche Funktion verwenden Sie hier? Welche Werte muss man für die Parameter einsetzen?

***Tipp:***<br>
- Die Datei sr_aufg_3d_xy.txt ist nur aus technischen Gründen im .txt-Format abgespeichert. Sie muss nicht in das Notebook eingelesen werden, sondern kann auch mit Microsoft Word, WordPad oder dem Editor gelesen werden.
- Es treten (je nach Matrikelnummer) Formulierungen wie "**höchstens** 9-mal", "**mindestens** 9-mal" oder "**genau** 9-mal" auf. Verwechseln Sie diese nicht!

In [7]:
from scipy.stats import binom

# Erstellung der Hypergeometrischen Verteilung
Tippfeld = hypergeom(70, 9, 20)

#Wahrscheinlichkeiten berechnen für 1,2,3,4 richtige, bei denen man Verlust macht
p=0
for i in range(1,5): # i in der range 1 bis 5, da man bei 0 keinen Verlust macht und 5, da es 5 Indizes gibt
    p += Tippfeld.pmf(i)

#Erstellen der Binominalverteilung
r = binom(12,p)

#Wahrscheinlichkeit für genau 9 mal Verlust bei 12 Spielen
wahrscheinlichkeit = r.pmf(9)

wahrscheinlichkeit.round(4)

0.0968

### Antwort:
- Die Wahrscheinlichkeit bei 12 Spielen genau 9 mal einen Verlust von 2 Euro zu haben, liegt bei 0.0968

- zuerst verwende ich die Funktion hypergeom(70, 9, 20) für die Hypergeometrische Verteilung, bei einem 70 Feld mit 20 Ziehungen und einem Tippfeld mit 9 kreuzen
- danach verwende ich die Funktion Tippfeld.pmf(i) für die Berechnung der Wahrscheinlichkeiten für 1, 2, 3, 4 richtige und addiere alle zusammen
- Die Wahrscheinlichkeiten p werden mit der Anzahl der Spiele 12 in der Funktion binom(12, p) verrechnet
- Mit der Funktion r.pmf(9) erhalte ich dann die Wahrscheinlichkeit für genau 9 mal Verlust bei 12 Spielen
